src/forms/Button.jsx

```diff
@@ -72,24 +73,31 @@ class Button extends React.PureComponent {
 		);
 
 		return (
-			<button
-				className={classNames.button}
-				onClick={onClick}
-				role='button'
-				{...other}
-			>
-				{ icon ? iconChildren : children }
-			</button>
+			React.cloneElement(
+				wrapperEl,
+				{
+					className: classNames.button,
+					onClick: onClick,
+					...other
+				},
+				icon ? iconChildren : children
+			)
 		);
 	}
 }
 
+Button.defaultProps = {
+	htmlElement: 'button',
+	wrapperEl: <button />
```

Comment:
Might be worth a followup ticket to clean this up, but `cloneElement` should only be used as a last-ditch option - it's pretty expensive and can oftentimes be avoided.

In this case, the prop should be a ReactComponent, not a ReactElement, and the `defaultProp` should be `wrapperEl: 'button'`

Then, in `render`, you would do something like

```jsx
const Component = this.props.wrapperEl;

return (
  <Component className={classNames.button} onClick={onClick} {...other}>
    {icon ? iconChildren : children }
  </Component>
);
```

Reply:
Filed a follow up as WC-199